latest/index.bs

```diff
@@ -233,34 +231,893 @@ keys as specified below for discovering certain types of data, especially images
 - MUST contain the field "name" that gives the name for this dimension. The values MUST be unique across all "name" fields.
-- SHOULD contain the field "type". It SHOULD be one of "space", "time" or "channel", but MAY take other values for custom axis types that are not part of this specification yet.
+- SHOULD contain the field "type". It SHOULD be one of "array", "space", "time", "channel", "coordinate", or "displacement" but MAY take other values for custom axis types that are not part of this specification yet.
+- MAY contain the field "discrete". The value MUST be a boolean, and is `true` if the axis represents a discrete dimension.
 - SHOULD contain the field "unit" to specify the physical unit of this dimension. The value SHOULD be one of the following strings, which are valid units according to UDUNITS-2.
     - Units for "space" axes: 'angstrom', 'attometer', 'centimeter', 'decimeter', 'exameter', 'femtometer', 'foot', 'gigameter', 'hectometer', 'inch', 'kilometer', 'megameter', 'meter', 'micrometer', 'mile', 'millimeter', 'nanometer', 'parsec', 'petameter', 'picometer', 'terameter', 'yard', 'yoctometer', 'yottameter', 'zeptometer', 'zettameter'
     - Units for "time" axes: 'attosecond', 'centisecond', 'day', 'decisecond', 'exasecond', 'femtosecond', 'gigasecond', 'hectosecond', 'hour', 'kilosecond', 'megasecond', 'microsecond', 'millisecond', 'minute', 'nanosecond', 'petasecond', 'picosecond', 'second', 'terasecond', 'yoctosecond', 'yottasecond', 'zeptosecond', 'zettasecond'
 
 If part of [[#multiscale-md]], the length of "axes" MUST be equal to the number of dimensions of the arrays that contain the image data.
 
 
+"coordinateSystem" metadata {#coord-sys-md}
+--------------------------
+
+A "coordinate system" is a collection of "axes" / dimensions with a name. Every coordinate system:
+- MUST contain the field "name", that gives the space name. The values MUST be non-empty and unique.
+- MUST contain the field "axes", whose value is a valid set of "axes".
+
+<div class=example>
+
+```json
+{
+    "name" : "volume_micrometers",
+    "axes" : [
+        {"name": "x", "type": "space", "unit": "micrometer"},
+        {"name": "y", "type": "space", "unit": "micrometer"},
+        {"name": "z", "type": "space", "unit": "micrometer"}
+    ]
+}
+```
+</div>
+
+The order of the `"axes"` list matters and defines the index of each array dimension and coordinates for points in that
+coordinate system. For the above example, the `"x"` dimension is the first dimension. The "dimensionality" of a coordinate system
+is indicated by the length of its "axes" array.  The "volume_micrometers" example above is three dimensional (3D).
+
+
+### Array coordinate systems
+
+Every array has a default coordinate system whose parameters need not be explicitly defined.  Its name is the path to the array
+in the container, its axes have `"type":"array"`, are unitless, and have default "name"s. The ith axis has `"name":"dim_i"`
```

Comment:
I think the term array is heavily used throughout this document and is a bit overloaded. E.g. there is also an "affine array". There will be many other arrays within a store where a coordinate system doesn't make sense – like matrices in a table.

Would it be reasonable to instead call this a pixel coordinate system?

Reply:
Ah, I hadn't thought of pixels as specifically 2d. I think I have heard people refer to elements of a 3d image as pixels though. I've definitely seen that for color and time dimensions. [`hypervoxel`](https://stackoverflow.com/questions/16071467/word-that-encompasses-pixel-and-voxel)?

For `coordinate` and `displacement` the space is also defined by a zarr array (I think), but it's a different concept of array space.

Since it's specific to images, and refers to the native image space, the type of space could be `"image"`, `"image-native"`?

Reply:
I agree that overloading the meaning of `array` again can potentially be confusing.